libminifi/include/c2/C2Payload.h

```diff
@@ -79,21 +85,65 @@ enum Direction {
   RECEIVE
 };
 
-struct AnnotatedValue : state::response::ValueNode {
-  using state::response::ValueNode::ValueNode;
-  using state::response::ValueNode::operator=;
+class C2Value {
+ public:
+  friend std::ostream& operator<<(std::ostream& out, const C2Value& val);
+
+  C2Value() = default;
+  C2Value(const C2Value& other) {
+    (*this) = other;
+  }
+  C2Value(C2Value&&) = default;
+  template<typename T>
+  requires(std::constructible_from<state::response::ValueNode, T>)
+  C2Value(T&& value) { value_ = state::response::ValueNode{std::forward<T>(value)}; }  // NOLINT(runtime/explicit)
+  C2Value(const rapidjson::Value& json_value) {  // NOLINT(runtime/explicit)
+    value_.emplace<rapidjson::Document>();
+    get<rapidjson::Document>(value_).CopyFrom(json_value, get<rapidjson::Document>(value_).GetAllocator());
+  }
+  C2Value(rapidjson::Document&& json_doc) {  // NOLINT(runtime/explicit)
```

Comment:
Why not explicit? A rationale after the NOLINT expression would be helpful for readers IMO.

Reply:
done